ios/RCTMGL/MGLModule.m

```diff
@@ -0,0 +1,78 @@
+//  Copyright © 2017 Mapbox Inc. All rights reserved.
+//
+
+#import "MGLModule.h"
+#import "RCTMGLEventTypes.h"
+@import Mapbox;
+
+@implementation MGLModule
+
+RCT_EXPORT_MODULE();
+
+- (NSDictionary<NSString *, id> *)constantsToExport
+{
+    // style urls
+    NSMutableDictionary *styleURLS = [[NSMutableDictionary alloc] init];
+    [styleURLS setObject:[MGLStyle.streetsStyleURL absoluteString] forKey:@"Street"];
+    [styleURLS setObject:[MGLStyle.darkStyleURL absoluteString] forKey:@"Dark"];
+    [styleURLS setObject:[MGLStyle.lightStyleURL absoluteString] forKey:@"Light"];
+    [styleURLS setObject:[MGLStyle.outdoorsStyleURL absoluteString] forKey:@"Outdoors"];
+    [styleURLS setObject:[MGLStyle.satelliteStyleURL absoluteString] forKey:@"Satellite"];
+    
+    // event types
+    NSMutableDictionary *eventTypes = [[NSMutableDictionary alloc] init];
+    [eventTypes setObject:RCT_MAPBOX_EVENT_TAP forKey:@"MapClick"];
+    [eventTypes setObject:RCT_MAPBOX_EVENT_LONGPRESS forKey:@"MapLongClick"];
+    [eventTypes setObject:RCT_MAPBOX_REGION_WILL_CHANGE_EVENT forKey:@"RegionWillChange"];
+    [eventTypes setObject:RCT_MAPBOX_REGION_IS_CHANGING forKey:@"RegionIsChanging"];
+    [eventTypes setObject:RCT_MAPBOX_REGION_DID_CHANGE forKey:@"RegionDidChange"];
+    [eventTypes setObject:RCT_MAPBOX_WILL_START_LOADING_MAP forKey:@"WillStartLoadingMap"];
+    [eventTypes setObject:RCT_MAPBOX_DID_FINISH_LOADING_MAP forKey:@"DidFinishLoadingMap"];
+    [eventTypes setObject:RCT_MAPBOX_DID_FAIL_LOADING_MAP forKey:@"DidFailLaodingMap"];
+    [eventTypes setObject:RCT_MAPBOX_WILL_START_RENDERING_FRAME forKey:@"WillStartRenderingFrame"];
+    [eventTypes setObject:RCT_MAPBOX_DID_FINSIH_RENDERING_FRAME forKey:@"DidFinishRenderingFrame"];
+    [eventTypes setObject:RCT_MAPBOX_DID_FINISH_RENDERING_FRAME_FULLY forKey:@"DidFinishRenderingFrameFully"];
+    [eventTypes setObject:RCT_MAPBOX_WILL_START_RENDERING_MAP forKey:@"WillStartRenderingMap"];
+    [eventTypes setObject:RCT_MAPBOX_DID_FINISH_RENDERING_MAP forKey:@"DidFinishRenderingMap"];
+    [eventTypes setObject:RCT_MAPBOX_DID_FINISH_RENDERING_MAP_FULLY forKey:@"DidFinishRenderingMapFully"];
+    [eventTypes setObject:RCT_MAPBOX_DID_FINISH_LOADING_STYLE forKey:@"DidFinishLoadingStyle"];
+    [eventTypes setObject:RCT_MAPBOX_FLY_TO_COMPLETE forKey:@"FlyToComplete"];
+    [eventTypes setObject:RCT_MAPBOX_SET_CAMERA_COMPLETE forKey:@"SetCameraComplete"];
+    
+    // user tracking modes
+    NSMutableDictionary *userTrackingModes = [[NSMutableDictionary alloc] init];
+    [userTrackingModes setObject:[NSNumber numberWithInt:MGLUserTrackingModeNone] forKey:@"None"];
+    [userTrackingModes setObject:[NSNumber numberWithInt:MGLUserTrackingModeFollow] forKey:@"Tracking"];
+    [userTrackingModes setObject:[NSNumber numberWithInt:MGLUserTrackingModeFollowWithHeading] forKey:@"Compass"];
+    [userTrackingModes setObject:[NSNumber numberWithInt:MGLUserTrackingModeFollowWithCourse] forKey:@"Navigation"];
```

Comment:
This is where things will get interesting, the iOS and Android SDKs have different names for the modes. Unless I am looking in the wrong place

Reply:
I ran into similar issues while working on a PR you recently merged. Android and iOS have different OfflinePackStates, so I had to create a method to normalize them into one. In that particular case, iOS had more states than Android, so it made sense to map them to the iOS states. In this particular case, I agree with @bsudekum. The iOS names are more descriptive. The actual result here isn't "compass" it's "follow the user with compass rotation." I also find "Tracking" vs "Compass" not very self-descriptive. While I know the answer to these questions, here is an example: If following the user is "tracking" then is "compass" not following the user and just rotating?

Ideally, this would be fixed upstream, but since it isn't, I think we should run with more descriptive, even if it means choosing our own names entirely for RN.

Reply:
I'll switch the names over. I agree we should be using more descriptive names.